### PR TITLE
cleanup Hexagon conv2d tests

### PR DESCRIPTION
**Pull request by adstraw**
Deletes unnecessary cases from the test_conv2d_blocked.py including cases covering logical `oihw` filter and packed `nhw8h8wc` packed input layouts.  Moved quite a lot of shared code to the infrastructure file reducing maintenance costs.  And, some other general cleanup items --- comments, variable names, etc.

### CODE REVIEW

**Comment by adstraw** [comment]
@csullivan and @cconvey please review

File: tests/python/contrib/test_hexagon/conftest.py [hunk 1]
```diff
@@ -1,37 +0,0 @@
...
-
-
-@tvm.testing.fixture
-def shape_oihw(out_channel, in_channel, kernel):
-    return (out_channel, in_channel, kernel, kernel)
-
-
-@tvm.testing.fixture
-def shape_oihw8i32o4i(out_channel, in_channel, kernel):
-    return get_packed_filter_layout(out_channel, in_channel, kernel, kernel)
```
Comment (csullivan) [inline tests/python/contrib/test_hexagon/conftest.py:37]: The goal being to check to see if the testing fixtures are the source of the `i386` flakiness?

Comment (adstraw) [reply tests/python/contrib/test_hexagon/conftest.py:37]: I removed the fixtures because of their effect in terms of the ordering of parameters on the pytest command lines which I personally find confusing.  I just realized, though, that the maxpool test in this same directory relies on these fixtures so I either need to remove that dependence or add back these fixtures.

File: tests/python/contrib/test_hexagon/infrastructure.py [hunk 2]
```diff
@@ -18,36 +18,43 @@
...
+    block_shape = get_block_shape()
     off_h, off_w, off_c = block_shape
     shape.append(ceildiv(shape_nhwc[1], off_h))
     shape.append(ceildiv(shape_nhwc[2], off_w))
-    if packed_C:
-        shape.append(ceildiv(shape_nhwc[3], off_c))
-        shape.extend(block_shape)
-    else:
-        shape.extend([off_h, off_w, shape_nhwc[3]])
+    shape.append(ceildiv(shape_nhwc[3], off_c))
```
Comment (cconvey) [inline tests/python/contrib/test_hexagon/infrastructure.py:44]: In this code we handle `ceildiv` as-is, but in the `get_packed_filter_shape` function (below) we cast its result to a Python `int` type.

Is there a good reason for doing it two different ways?  

Specifically: casting to `int` works when the runtime return type of `ceildiv` is `tvm.tir.expr.IntImm`.  But I'm wondering if we're okay with this test making that assumption. I.e., are we testing everything we _intend_ to test when we assume that the exact shapes are known at this point in time?

File: tests/python/contrib/test_hexagon/infrastructure.py [hunk 3]
```diff
@@ -18,36 +18,43 @@
 """ Hexagon testing infrastructure """
 
 import tvm
+from tvm import te
 import numpy
 
 
 def ceildiv(o, d):
     return tvm.tir.floordiv(o + d - 1, d)
```
Comment (cconvey) [inline tests/python/contrib/test_hexagon/infrastructure.py:26]: Should we add a comment that this function might only work as intended for _positive_ numbers?

tl;dr:
It's not obvious if this has intuitive behavior when `o` and/or `d` are negative. AFAIK `o` and `d` will only be positive when we're calling this, but someone might decide to promote this function to a broader role in TVM.

File: tests/python/contrib/test_hexagon/infrastructure.py [hunk 4]
```diff
@@ -18,36 +18,43 @@
...
-def get_packed_activation_layout(shape_nhwc, block_shape, packed_C=True):
+def get_block_shape():
+    return 8, 8, 32
+
+
+def get_filter_block_shape():
+    return 8, 32, 4
+
+
+def get_packed_shape(shape_nhwc):
```
Comment (cconvey) [inline tests/python/contrib/test_hexagon/infrastructure.py:37]: Since we have several competing versions of "shape" in this code, would it be helpful (for clarity) to rename `shape_nhwc` to `logical_shape_nhwc`?

File: tests/python/contrib/test_hexagon/infrastructure.py [hunk 5]
```diff
@@ -18,36 +18,43 @@
...
     return tvm.tir.floordiv(o + d - 1, d)
 
 
-def get_packed_activation_layout(shape_nhwc, block_shape, packed_C=True):
+def get_block_shape():
+    return 8, 8, 32
+
+
+def get_filter_block_shape():
+    return 8, 32, 4
```
Comment (cconvey) [inline tests/python/contrib/test_hexagon/infrastructure.py:34]: It's not obvious why these particular numbers are chosen. Should there be comments explaining the reasons for using them?

**Review by cconvey — CHANGES_REQUESTED** [review]

File: tests/python/contrib/test_hexagon/infrastructure.py [hunk 6]
```diff
@@ -18,36 +18,43 @@
...
+from tvm import te
 import numpy
 
 
 def ceildiv(o, d):
     return tvm.tir.floordiv(o + d - 1, d)
 
 
-def get_packed_activation_layout(shape_nhwc, block_shape, packed_C=True):
+def get_block_shape():
```
[the remark this answers is not in the document]
Comment (adstraw) [reply tests/python/contrib/test_hexagon/infrastructure.py:29]: Added a basic comment:  "defines inner block shape: 8h8w32c"

File: tests/python/contrib/test_hexagon/infrastructure.py (hunk 2 above)
Comment (adstraw) [reply tests/python/contrib/test_hexagon/infrastructure.py:44]: Indeed, casting to `int` is a means of converting from `tvm.tir.expr.IntImm` to `int`.  I am not sure why it works differently for `get_packed_shape` (no cast) vs. `get_packed_filter_shape` (requires cast).  Also not sure about your question RE knowing exact shapes.  More investigation required.

NOTE:  I did change `get_packed_filter_shape` to take a shape instead of a list of dimensions to match the API of `get_packed_shape`.

File: tests/python/contrib/test_hexagon/infrastructure.py (hunk 3 above)
Comment (adstraw) [reply tests/python/contrib/test_hexagon/infrastructure.py:26]: I added some asserts to catch cases where `o` or `d` are less than zero.

File: tests/python/contrib/test_hexagon/infrastructure.py (hunk 5 above)
Comment (adstraw) [reply tests/python/contrib/test_hexagon/infrastructure.py:34]: Added a basic comment: "defines inner filter block shape: 8i32o41"

File: tests/python/contrib/test_hexagon/infrastructure.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (adstraw) [reply tests/python/contrib/test_hexagon/infrastructure.py:37]: Renamed to `logical_shape_nhwc`.  Also renamed other shape parameters in the infrastructure file for clarity.

In regards to padding the role of this function is to convert the post-padded logical shape into a physical packed shape.  I think the intent of this function is clear, now, given the variable name change.  Also added comments to this effect.

Comment (adstraw) [reply tests/python/contrib/test_hexagon/infrastructure.py:37]: Done.

**Review by cconvey — APPROVED** [review]

**Review by csullivan — APPROVED** [review]
LGTM

**Review by tmoreau89 — APPROVED** [review]
LGTM

**Comment by tmoreau89** [comment]
Thank you @adstraw @cconvey @csullivan ; the PR has been merged!
